src/store.c

```diff
@@ -351,14 +352,18 @@ int pthreads_store_write(zval *object, zval *key, zval *write) {
 				normal refcounting, we'll just never read the reference
 			*/
 			rebuild_object_properties(&threaded->std);
+			
+			pthreads_store_sync(object);
```

Comment:
It's not immediately clear to me why this is needed - is it related to the memory leaks you're fixing? And do you have a test case for this?